### PR TITLE
Refactor steps processor to remove use of @params

### PR DESCRIPTION
**Pull request by mloughran**
This makes it easier to confirm that use of user submitted data within the steps processor is safe.

The previous design allowed unsafe practices, and encouraged defensive code (e.g. accessing the prison_id from the params hash rather than the type checked version from the prisoner step).

See also https://github.com/ministryofjustice/prison-visits-2/pull/245, which inspired this change.

### CODE REVIEW

**Comment by mloughran** [comment]
(tests will pass when I rebase onto #12)

File: app/services/steps_processor.rb [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
+
   def initialize(params, locale)
-    @params = params
-    @steps = load_steps
+    @steps = load_steps(params)
     @locale = locale
+
+    # Compute now to avoid keeping @params
+    @steps_submitted = STEP_NAMES.select { |s| params.has_key?(s) }
+    @review_step = STEP_NAMES.find { |s| s.to_s == params[:review_step].to_s }
```
Comment (alan) [inline app/services/steps_processor.rb]: You kept the `.find` at the end?

File: app/services/steps_processor.rb [hunk 2]
```diff
@@ -1,16 +1,21 @@
 class StepsProcessor
+  STEP_NAMES = %i[ prisoner_step visitors_step slots_step confirmation_step ]
```
Comment (alan) [inline app/services/steps_processor.rb]: They aren't names, they are the steps themselves, how about calling it `STEPS`?

**Comment by alan** [comment]
👍

**Pull request merged**